File: src/main/java/space/accident/api/interfaces/tileentity/redstone/IRedStoneTileEntity.java
package space.accident.api.interfaces.tileentity.redstone;

/**
 * This File has just internal Information about the RedStone State of a TileEntity
 */
public interface IRedStoneTileEntity extends IRedStoneEmitter, IRedStoneReceiver {
	/**
	 * enables/disables RedStone Output in general.
	 */
	void setGenericRedstoneOutput(boolean aOnOff);
	
	/**
	 * Causes a general Block update.
	 * Sends nothing to Client, just causes a Block Update.
	 */
	void issueBlockUpdate();
}

File: src/main/kotlin/space/accident/api/interfaces/te/IRedstoneTileEntity.kt
package space.accident.api.interfaces.te

/**
 * This File has just internal Information about the Redstone State of a TileEntity
 */
interface IRedstoneTileEntity : IRedstoneEmitter, IRedstoneReceiver {
    /**
     * enables/disables Redstone Output in general.
     */
    fun setGenericRedstoneOutput(aOnOff: Boolean)

    /**
     * Causes a general Block update.
     * Sends nothing to Client, just causes a Block Update.
     */
    fun issueBlockUpdate()
}